waifu2x.ts

```diff
@@ -156,35 +161,51 @@ export default class Waifu2x {
         if (process.platform === "linux") program = `cd "${absolute}" && ./cwebp`
         let command = `${program} -q ${quality} "${source}" -o "${dest}"`
         const child = child_process.exec(command)
+        Waifu2x.addProcess(child)
+        //TODO add error handling
         await new Promise<void>((resolve, reject) => {
-            child.on("close", () => resolve())
+            child.on("close", () => {
+                Waifu2x.removeProcess(child)
+                resolve()
+            })
         })
         return dest
     }
-
+    private static addProcess = (process: child_process.ChildProcess) => {
+        Waifu2x.processes.push(process)
+    }
+    private static removeProcess = (process: child_process.ChildProcess) => {
+        Waifu2x.processes = Waifu2x.processes.filter((p) => p.pid !== process.pid)
+    }
     public static convertFromWebp = async (source: string, dest: string, webpPath?: string) => {
         const absolute = webpPath ? path.normalize(webpPath).replace(/\\/g, "/") : path.join(__dirname, "../webp")
         let program = `cd "${absolute}" && dwebp.exe`
         if (process.platform === "darwin") program = `cd "${absolute}" && ./dwebp.app`
         if (process.platform === "linux") program = `cd "${absolute}" && ./dwebp`
         let command = `${program} "${source}" -o "${dest}"`
         const child = child_process.exec(command)
+        Waifu2x.addProcess(child)
         let error = ""
         await new Promise<void>((resolve, reject) => {
             child.stderr.on("data", (chunk) => error += chunk)
-            child.on("close", () => resolve())
+            child.on("close", () => {
+                Waifu2x.removeProcess(child)
+                resolve()
+            })
         })
         if (error.includes("animated WebP")) return Promise.reject(error)
         return dest
     }
 
     public static upscaleImage = async (source: string, dest?: string, options?: Waifu2xOptions, progress?: (percent?: number) => void | boolean) => {
-        if (!options) options = {}
-        if (!dest) dest = "./"
-        if (options.rename === undefined) options.rename = "2x"
+        options = {...options} || {}
+        dest = dest ?? "./"

```

Comment:
For options, I don't really see what you changed here, you just wrote it in a less visually pleasing way.

For dest, it might cause problems if they pass in a empty string (though I haven't tested it), I think it's better if you replace it for all falsy values and not just undefined or null.

I would rather you don't change these lines and keep it the way it was (this goes for all the other times you changed it to `options = {...options} || {}`.

Reply:
Wait actually that wouldn't even work thinking about it, since {...undefined} is an empty object, but I'd keep the clone because lines below modify the options object which would be undesirable since it changes it as reference of the object that called it, which shouldn't be a correct behaviour. 
I'll change back the rest tho, tell me if the object clone/spread is fine

Reply:
If changing by reference is a problem for you then sure, you can keep the object clone.